x-pack/plugins/canvas/public/apps/workpad/routes.js

```diff
@@ -11,6 +11,7 @@ import { setWorkpad } from '../../state/actions/workpad';
 import { setAssets, resetAssets } from '../../state/actions/assets';
 import { gotoPage } from '../../state/actions/pages';
 import { getWorkpad } from '../../state/selectors/workpad';
+import { setReadOnlyUser } from '../../state/actions/transient';
```

Comment:
Currently there are only read or all privileges in Kibana, so I don't believe it's necessary to track read permissions since every user at a baseline has read privileges. It should be sufficient to just track if the user has write permissions.